src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +215,21 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param types      The types of logs being requested.
+ * @param timeout    The timeout for getting the logs.  If the timeout expires, completion
+ *                   will be called with whatever logs have been retrieved by that point
+ *                   (which might be none, and might include partial logs).
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to pass in the file paths for the requested logs.
+ */
+- (void)getDiagnosticLogsOfTypes:(MTRDiagnosticLogTypes)types
```

Comment:
we would also need the protocol supports and a list of file designators for BDX if we want the client to specify that. I think we should add that. I will add it when I address any more comments from you.

Reply:
The client should not need to do anything with file designators.

Why would the client ever do anything with protocol supports here?